skbio/io/format/embl.py

```diff
@@ -0,0 +1,1568 @@
+def _parse_source(lines):
+    '''Parse single SOURCE field.
+    '''
+
+    # parsed reference will be placed here
+    res = {}
+
+    # define a section splitter with _embl_yield_section function defined in
+    # this module
+    section_splitter = _embl_yield_section(lambda line: _get_embl_key(line),
+                                           skip_blanks=True, strip=False)
+
+    # now itereta along sections (lines of the same type)
+    for section in section_splitter(lines):
+        # this function append all data in the same keywords. A list of lines
+        # as input (see skbio.io.format._sequence_feature_vocabulary)
+        label, data = _embl_parse_section_default(
+            section, join_delimiter=' ', return_label=True)
+
+        res[label] = data
+
+    # return translated keys
+    return _translate_keys(res)
+
+
+def _serialize_source(header, obj, indent=5):
+    '''Serialize SOURCE.
+
+    Parameters
+    ----------
+    header: section header
+    obj : dict
+    indent : indent length
+    '''
+
+    source = []
+
+    # treat taxonomy and all others keys
+    for key in ["ORGANISM", "taxonomy", "organelle"]:
+        # get data to serielize
+        data = obj.get(key)
+
+        # if key is not defined (eg. organelle, continue)
+        if data is None:
+            continue
+
+        # get embl key for my key (eg, taxonomy -> OC)
+        embl_key = REV_KEYS_TRANSLATOR.get(key, key)
+
+        # get an embl wrapper
+        wrapper = _get_embl_wrapper(embl_key, indent)
+
+        # define wrapped string
+        source += wrapper.wrap(data)
+
+    # now define a string and add a final "\n"
+    s = "\n".join(source) + "\n"
+
+    # and return it
+    return s
+
+
+def _parse_sequence(lines):
+    '''Parse the sequence section for sequence.'''
+
+    # when reading a feature-level-products accession, features are relative
+    # to parent accession, so we need to parse accessions like
+    # LK021130.1:74067..75610:rRNA to model a table feature like
+    # FT   rRNA            LK021130.1:74067..75610
+    # FT                   /gene="16S"
+    # FT                   /product="16S rRNA"
+    # FT                   /note="16S rRNA subunit (checked and believed
+    # FT                   to be right,based on 454- and PacBio-sequencing)"
+    # I could express features by resizing sequence, however the resulting
+    # sequence will be different than data read. I decide to mantain
+    # sequences indentical to read values, and to discard features
+    # for such entries
+
+    # result array
+    sequence = []
+
+    for line in lines:
+        # ignore record like:
+        # SQ   Sequence 275 BP; 64 A; 73 C; 88 G; 50 T; 0 other;
+        if line.startswith('SQ'):
+            continue
+
+        # remove the numbers inside strings. revome spaces around string
+        line = ''.join([i for i in line.strip() if not i.isdigit()])
+
+        # remove space from sequence
+        line = line.replace(" ", "")
+
```

Comment:
The problem is that embl record can have numbers or not for example, see [uniprot](https://github.com/bunop/scikit-bio/blob/issue-1483/skbio/io/format/tests/data/embl_uniprot_record#L43) and [embl](https://github.com/bunop/scikit-bio/blob/issue-1483/skbio/io/format/tests/data/embl_single_record#L105) records in unittest: The first has no numbers, the second yes. I don't remeber if I found another record in ENA with no line numbers, but I wrote this line to deal with both record types.